Add Contentful webhook function.

DFurnes opened the pull request:
This pull request adds another function to our GraphQL app, running the `webhook.handler` handler.

References DoSomething/graphql#96.

weerd commented on applications/graphql/main.tf:
```diff
@@ -97,20 +97,41 @@ module "app" {
   domain = "${var.domain}"
 
-  functions     = ["${module.app.arn}"]
+  functions     = ["${module.app.arn}", "${module.contentful_webhook.arn}"]
```
what's an `arn`? 🤔

DFurnes replied:
Amazon's [silly name for IDs](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html). 🤷‍♂️

weerd replied:
ah! 🤪

DFurnes commented:
Alas, this became a wee bit more complicated –

1. Added permissions so that both Lambda functions can access the DynamoDB table in b3af523. This was a relatively straightforward change to make the `role` argument into a `roles` array in the `dynamodb_cache` component (which is only used here).
2. Fixed a Terraform 0.11 issue with dynamic lists used for the [`count` argument](https://www.terraform.io/docs/configuration/resources.html#count-multiple-resource-instances) by passing a (manually maintained, but "validated") count integer argument alongside it. This hack can be removed once we're on Terraform 0.12.

DFurnes commented:
Confirmed that this PR now makes a ton of new resources for the new Lambdas & their permissions/routes, but does **not** delete or change any existing resources.

```
Plan: 74 to add, 0 to change, 0 to destroy.
```

DFurnes commented:
I ran into an interesting snag yesterday – by default, the API Gateway deployment [doesn't update](https://github.com/hashicorp/terraform/issues/6613) when routes change (and so those routes are not accessible by real traffic, even though they show up in the AWS console).

It seems like this can be fixed by setting a `variable` or `stage_description` on the deployment, which forces a rebuild. I tried that out yesterday and ran into conflicts with not being able to delete an active stage (reasonable), since Terraform always tries to delete an existing resource before building its replacement.

This (hopefully) can be fixed in d6e3d2e with the [`create_before_destroy`](https://www.terraform.io/docs/configuration/resources.html#lifecycle-lifecycle-customizations) lifecycle argument, which I've added in d6e3d2e.

DFurnes commented:
Alright, confirmed that this is working (with the default "hello world" function), and the new deployment is automatically created without issues across the board. 🕺 

```
curl -X POST https://graphql-dev.dosomething.org/contentful
<p>Hello world!</p>
```
